src/UserList.js

```diff
@@ -1,34 +1,87 @@
 import React from 'react';
-import usersData from './data/users.json';
-import User from './components/user.js';
+import User from './components/User.js';
 import './css/App.css';
+import UserStore from './stores/UserStore';
+import UserAction from './actions/UserAction';
 
 
 export default class UserList extends React.Component {
     constructor (props) {
         super(props);
 
-        this.state = {
-            wantAdd: false
-        };
+        this.state = this.getAppState();
 
         this.handleClickAddUser = this.handleClickAddUser.bind(this);
     }
 
+    getAppState () {
+        return {
+            wantAdd: false,
+            users:   UserStore.getUsers()
+        };
+    }
+
     handleClickAddUser () {
         this.setState({ wantAdd: !this.state.wantAdd });
     }
 
+    handleClickSubmit () {
+        let address = {};
+        if(document.getElementById('address')) {
+            address = {
+                city:    document.getElementById('city').value ,
+                street:  document.getElementById('street').value,
+                suite:   document.getElementById('suite').value,
+                zipcode: document.getElementById('zipcode').value
+            }
+        }
+        let company = {};
+        if(document.getElementById('company')) {
+            company = {
+                name:        document.getElementById('nameCompany').value,
+                catchPhrase: document.getElementById('name').value,
+                bs:          document.getElementById('bs').value
+            }
+        }
+        let user = {
+            name:    document.getElementById('name').value,
+            phone:   document.getElementById('phone').value,
+            email:   document.getElementById('email').value,
+            website: document.getElementById('website').value,
+            address: address,
+            company: company
+        };
+        UserAction.addNewUser(user);
+    }
+
+    componentDidMount () {
+        UserStore.addChangeListener(this.onChange);
```

Comment:
We usully do not create a separate methods to add lister to a particalar event of the store because in this case we have to have a pair of such methods for each event.
It's better to use `store.on/store.off` or `store.addListener/store.removeListener` directly